core/commands/resolve.go

```diff
@@ -82,12 +82,15 @@ Resolve the value of an IPFS DAG path:
 		// the case when ipns is resolved step by step
 		if strings.HasPrefix(name, "/ipns/") && !recursive {
 			p, err := n.Namesys.ResolveN(req.Context(), name, 1)
+
+			if p != "" {
```

Comment:
I dont like treating this as a special case... If we call the path resolved, the error should be nil.